From daaf7489136d2e0ad019c9e6883b3d67c10902df Mon Sep 17 00:00:00 2001
From: Martin Kaiser <martin.kaiser@codasip.com>
Date: Fri, 22 Aug 2025 14:41:06 +0200
Subject: [PATCH] trace-cmd: fix container_of to work with cheri

Cast the member pointer to intptr_t to preserve the capability metadata.

Upstream-Status: Inappropriate [Cheri-Specific]
---
 tracecmd/include/list.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tracecmd/include/list.h b/tracecmd/include/list.h
index fa0de6df..98a1a769 100644
--- a/tracecmd/include/list.h
+++ b/tracecmd/include/list.h
@@ -6,8 +6,10 @@
 #ifndef __LIST_H
 #define __LIST_H
 
+#include <stdint.h>
+
 #define offset_of(type, field)		__builtin_offsetof(type, field)
-#define container_of(p, type, field)	(type *)((long)p - offset_of(type, field))
+#define container_of(p, type, field)	(type *)((intptr_t)p - offset_of(type, field))
 
 struct list_head {
 	struct list_head *next;
